### DISCUSSION TITLE
For mixed convection for Pipe flow

### DISCUSSION THREAD

**Discussion by Madhurjya814** [xcompact3d/Incompact3d#327] [General]
Hi,
I want to do dns simulation in a pipe for mixed convection case as giving heat flux at the wall of the pipe. What are the modifications to be done and in which source files for giving constant heat flux boundary condition and adding the buoyancy term in the momentum equation?
Thanks

**Comment by mathrack** [comment]
Hello,
The current IBM can be used to impose the value at the wall. We plan to include later the developments allowing to impose the heat flux and to perform fluid-solid thermal coupling. See https://www.sciencedirect.com/science/article/pii/S0021999123002772
However, due to time constraints this is not available currently.
Cheers

**Comment by Madhurjya814** [comment]
Thanks for the reply.
I just wanted to know that if I have to implement the heat flux condition and buoyancy by myself, then is it possible to do so?
Thanks

**Reply by mathrack** [reply]
It will be difficult but not impossible

**Reply by Madhurjya814** [reply]
Sir, can you give me some guidance how can I give heat flux at the wall and add buoyancy force so that I can use for mixed convection case. What are the files that I have to understand and modify?
Thanks.

**Reply by mathrack** [reply]
Try to understand how the IBM works currently. This will lead you directly to the files you will have to modify.
Cheers

**Reply by Madhurjya814** [reply]
Thanks for your valuable reply Sir. I will try my best.

**Reply by Madhurjya814** [reply]
Hi,
I have found this paper https://www.sciencedirect.com/science/article/pii/S0142727X24003655 on turbulent mixed convection inside pipe. Here it is written that Xcompact3d software is used. It has not mentioned that the solver has been modified for the use. Is that mean that for pipe flow, mixed convection is already there in the software?
Thanks.

**Reply by mathrack** [reply]
In the publication, we are using a modified version of xcompact3d.

The code on github does not include buoyancy effects, you need to add a source term to include buoyancy.

The code on github can be used to have an imposed value with immersed boundaries. Some modifications are needed to have an imposed heat flux with immersed boundaries.

**Reply by Madhurjya814** [reply]
Hello Sir,
continuing the previous discussion, I would like to know what are src files that need to be modified and which subroutines to be modified in order to include the buoyancy force term. Thanking you.

**Reply by mathrack** [reply]
Hello,

The momentum rhs is computed in the subroutine `momentum_rhs_eq`. The temperature rhs is computed in `scalar_transport_eq`. They are available in the file `transeq.f90`.

Cheers
